middleware/recon.go

```diff
@@ -83,8 +83,16 @@ func DumpReconCache(reconCachePath string, source string, cacheData map[string]i
 			reconData[key] = item
 		}
 	}
-	newdata, _ := json.Marshal(reconData)
-	return common.WriteFileAtomic(reconFile, newdata, 0600)
+	newdata, err := json.Marshal(reconData)
+	if err != nil {
+		return err
+	}
+	f, err := fs.NewAtomicFileWriter(reconCachePath, reconCachePath)
```

Comment:
I'll add it.  I think we didn't need it since there's nowhere for the code to go, but I guess it could panic or something else.